frontend/deb/handle_debroot.go

```diff
@@ -0,0 +1,148 @@
+package deb
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/pkg/errors"
+	"golang.org/x/exp/maps"
+)
+
+func DebrootHandler(target string) frontend.BuildFunc {
+	return func(ctx context.Context, client gwclient.Client, spec *dalec.Spec) (gwclient.Reference, *frontend.Image, error) {
+		st, err := Debroot(spec, llb.Scratch(), target, "")
+		if err != nil {
+			return nil, nil, err
+		}
+
+		sOpt, err := frontend.SourceOptFromClient(ctx, client)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		sm, err := dalec.Sources(spec, sOpt)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		for k, src := range sm {
+			sm[k] = llb.Scratch().File(llb.Copy(src, "/", k, dalec.WithCreateDestPath()))
```

Comment:
Why are we doing copies here and not mounts?

Reply:
Because we aren't running anything just pulling out the content for the debroot that will be exported to the client.